File: factorial/lua_factorial/factorial.lua
function factorial(n)
  if ( n == 0 ) then
    return 0
  elseif( n == 1 ) then
    return 1
  else
    return n * factorial(n - 1)
  end
end

function main()
  for idx_i=1,14,1
  do
    print("factorial(" .. idx_i .. ")=" .. tostring(factorial(idx_i)))
  end
end

main()

File: factorial/scala_factorial/factorial.scala
object factorial {
  def factorial(n:Int):Int = {
  if (n==0)
    return 1
  else
    return n * factorial(n-1)
  }

  def main(args: Array[String]): Unit = {
    for( idx_i <- 1 to 13) {
      println( "factorial(" + idx_i + ")=<" + factorial(idx_i) + ">")
    }
  }
}
